### DISCUSSION TITLE
How to use an ObjectModel from a module on a selector field

### DISCUSSION THREAD

**Discussion by alicedoe** [PrestaSafe/prettyblocks#280] [Q&A]
Hi,

I need to call a collection from a blog module that has a SimpleBlogPost class extending ObjectModel. Is this possible?

Currently, if I use the following:
```PHP
'blog' => [
	'type' => 'selector',
	'label' => 'Choose a product',
	'collection' => 'SimpleBlogPost',
	'default' => 'default value',
	'selector' => '{id} - {name}'
]
```
Prettyblocks crashes with "Fatal error: Uncaught PrestaShopException: Field id_simpleblogpost not found in class SimpleBlogPost" in the module the primary field is id_simpleblog_post, I've added in Fieldcore->_getCollection
```PHP
if ( $collectionName == 'SimpleBlogPost' ) { $primaryField = 'id_simpleblog_post'; }
```
$c = new \PrestaShopCollection($collectionName, $this->id_lang); returns something
but $object = $c->where($primary, '=', (int) $id)->getFirst(); returns false

So is it possible to use an ObjectModel from a module on a selector field or only native prestashop ObjectModel ?

Best regards, Alice

**Comment by PrestaSafe** [comment]
Hi, instead of `{id}` can you tried to put `{id_simpleblog_post}` in your `'selector'` field ? 

like that: 
`'selector' => '{id_simpleblog_post} - {name}'`

**Comment by alicedoe** [comment]
Hi,

thanks for your answer

```PHP
'blog' => [
        'type' => 'selector',
        'label' => 'Choose a blog post',
        'collection' => 'SimpleBlogPost',
        'default' => 'default value',
        'selector' => '{id_simpleblog_post} - {title}'
],
```
there si no name in SimpleBlogPost in so I used {title}

```PHP
    public function _getCollection($id, $collectionName, $primaryField = null)
    {
		
		if ( $collectionName == 'SimpleBlogPost' ) { $primaryField = 'id_simpleblog_post'; }
        $c = new \PrestaShopCollection($collectionName, $this->id_lang);
        $primary = $primaryField ?? 'id_' . \Tools::strtolower($collectionName);

        $object = $c->where($primary, '=', (int) $id)->getFirst();
		\PrestaShopLogger::addLog($collectionName . print_r($object, true), 1, null, 'PrettyBlocks');
        if (!\Validate::isLoadedObject($object)) {
            return false;
        }
        $objectPresenter = new ObjectPresenter();
        return $objectPresenter->present($object);
    }
```
$object is still empty for SimpleBlogPost

I have no error but it's impossible to find a blog post : 
![blog_post](https://github.com/user-attachments/assets/f019b328-1886-47e8-b204-c0773dbefb64)

**Comment by PrestaSafe** [comment]
Sorry, i need to search deeper, i was sure to put an option `primary` for this field, but, i don't see it. 
I will try to implemented it soon.

**Comment by alicedoe** [comment]
Hi, was this implementation included in version 3.2.0?
